### ISSUE TITLE
`@Transactional` not working with `@Nested` and `@QuarkusTest`

### ISSUE THREAD

**Issue by knutwannheden** [quarkusio/quarkus#29978]
### Describe the bug

It appears that in `@Nested` JUnit `@QuarkusTest`s the CDI interceptors don't get executed. In any case the following nested test case annotated with `@Transactional` fails (whereas the equivalent top-level test succeeds):

```java
@QuarkusTest
class MyRepoTest {

    @Test
    @Transactional
    void test() {
        assertTrue(QuarkusTransaction.isActive());
    }

    @Nested
    class NestedTest {
        @Test
        @Transactional
        void nestedTest() {
            assertTrue(QuarkusTransaction.isActive());
        }
    }
}
```

### Expected behavior

It would be nice if this could be made to work. But since nested JUnit5 tests so often seem to cause trouble, this may not be possible. If so, it would be nice with a compile-time or runtime error in this situation and / or a hint in the documentation.

### Actual behavior

The `@Transactional` CDI interceptor on the `@Nested` test simply appears to be ignored.

### How to Reproduce?

For a full reproducer (including test in description) see this attachment: [nested-transaction.zip](https://github.com/quarkusio/quarkus/files/10269023/nested-transaction.zip)

### Quarkus version or git rev

2.15.0.Final

**Comment by geoand** [comment]
Sounds like a duplicate of https://github.com/quarkusio/quarkus/issues/20034, right?

**Comment by knutwannheden** [comment]
Yes, it does. Maybe Quarkus could at runtime check the annotations and fail with an error, if any of them has a CDI interceptor registered against it?

**Comment by geoand** [comment]
Perhaps, I have not looked into it
